contrib/auto-render/auto-render.js

```diff
@@ -25,10 +25,11 @@ const renderMathInText = function(text, delimiters) {
         } else {
             const span = document.createElement("span");
             const math = data[i].data;
+            // Override any display mode defined in the settings with that
+            // defined by the text itself
+            renderOptions.displayMode = data[i].display;
```

Comment:
As someone who has been bitten by this in the past (called function modifying callee's object), I agree.